data-prepper-plugins/encryption-plugin/src/main/java/org/opensearch/dataprepper/plugins/encryption/EncryptionSupplier.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.encryption;
+
+import org.opensearch.dataprepper.model.encryption.EncryptionEngine;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class EncryptionSupplier {
```

Comment:
This is the core class provided by the plugin. It should probably be an `interface` so that the plugin can define the implementation independently of the callers.

You can create a package-protected class that you hide from the plugin consumers.